admin/class-parcelpro-shipping.php

```diff
@@ -216,6 +216,31 @@ public function calculate_shipping($package = array())
             }
 
             foreach ($this->services as $carrier_name => $carrier) {
+                $lowercaseCarrier  = strtolower($carrier_name);
+                /** @var DateTimeImmutable $shipping_time */
+                $shipping_time = new DateTimeImmutable();
+                $rawCutoffTime = $this->get_option('parcelpro_forecast_settings_' . $lowercaseCarrier . '_cutoff_time');
+                $isBeforeCutoffTime = $this->isBeforeLastShippingTime($rawCutoffTime);
+
+                if (!$isBeforeCutoffTime) {
+                    $shipping_time = $shipping_time->add(new DateInterval('P1D'));
+                }
+
+                // Fetch expected delivery day
+                $delivery_expected = $this->api->getDeliveryDate(
+                    $carrier_name,
+                    $shipping_time,
+                    $package['destination']['postcode']
+                );
+                $formattedDeliveryDate = '';
+                $is_enabled = $this->get_option(
+                    'parcelpro_forecast_settings_' . $lowercaseCarrier . '_show_delivery_date'
+                );
+                if ($delivery_expected && $is_enabled === 'yes') {
```

Comment:
We could consider moving the `is_enabled` check around the fetching and calculation logic for a (small) optimization - mostly to prevent the unnecessary API calls. Although I'm also nog necessarily against leaving it as-is.